src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -211,6 +332,91 @@ private IFeatureFilterMetadata GetFeatureFilterMetadata(string filterName)
             return filter;
         }
 
+        private IFeatureVariantAssignerMetadata GetFeatureVariantAssignerMetadata(string assignerName)
+        {
+            const string assignerSuffix = "assigner";
+
+            IFeatureVariantAssignerMetadata assigner = _assignerMetadataCache.GetOrAdd(
+                assignerName,
+                (_) => {
+
+                    IEnumerable<IFeatureVariantAssignerMetadata> matchingAssigners = _variantAssigners.Where(a =>
+                    {
+                        Type assignerType = a.GetType();
+
+                        string name = ((AssignerAliasAttribute)Attribute.GetCustomAttribute(assignerType, typeof(AssignerAliasAttribute)))?.Alias;
+
+                        if (name == null)
+                        {
+                            name = assignerType.Name;
+                        }
+
+                        return IsMatchingMetadataName(name, assignerName, assignerSuffix);
+                    });
+
+                    if (matchingAssigners.Count() > 1)
+                    {
+                        throw new FeatureManagementException(FeatureManagementError.AmbiguousFeatureVariantAssigner, $"Multiple feature variant assigners match the configured assigner named '{assignerName}'.");
+                    }
+
+                    return matchingAssigners.FirstOrDefault();
+                }
+            );
+
+            return assigner;
+        }
+
+        /// <summary>
+        /// Trims a suffix from a name if necessary
+        /// </summary>
+        /// <param name="name">The name to trim.</param>
+        /// <param name="suffix">The possible suffix that may need trimming.</param>
+        /// <returns></returns>
+        private static string GetTrimmedName(string name, string suffix)
+        {
+            Debug.Assert(name != null);
+            Debug.Assert(suffix != null);
+
+            if (name.EndsWith(suffix, StringComparison.OrdinalIgnoreCase))
+            {
+                name = name.Substring(0, name.Length - suffix.Length);
+            }
+
+            return name;
+        }
+
+        private static bool IsMatchingMetadataName(string metadataName, string desiredName, string suffix)
+        {
+            //
+            // Support matching with suffix, i.e. "CustomFilter"
+            // Or without, i.e. "Custom"
+            return desiredName.EndsWith(suffix, StringComparison.OrdinalIgnoreCase) ?
+                IsMatchingMetadataName(metadataName, desiredName) :
+                IsMatchingMetadataName(GetTrimmedName(metadataName, suffix), desiredName);
```

Comment:
It will return true when, for example,
- desiredName == "MyProduct.Foo"
- metadataName == "MyProduct.FooFilter"

But based on your current design, we should return false (when namespaces are included, they must be exact match).

Reply:
I don't see a problem with those matching. The namespace matching mechanism is intended for different filter providers to be able to distinguish themselves. I don't think we need to break the matching behavior with/without the suffix.

Reply:
I see. It's going to be a breaking change. Probably not worth it. Stay the way it is then :)